core/selection.js

```diff
@@ -23,10 +23,7 @@
 		if ( ranges.length === 0 ) {
 			return false;
 		}
-		// It's not table selection when selected node is a widget (#1027).
-		if ( isWidget( ranges[ 0 ].getEnclosedNode() ) ) {
-			return false;
-		}
+
```

Comment:
This change is not related to this fix. We try to avoid changing things unrelated to the current fix in the same PR. I know it doesn't align with the proposed solution and it may seem tempting to move it together with proposed fix, however:

* This code is related to different fix which is similar but required a different approach (especially that it is related to widgets), so the assumption that things will work the same as for this fix may be incorrect.
* The solution was reviewed, accepted and tested already (some time ago) so changing it will require repeating the whole procedure from the previous issue perspective.